utils/autolispext.i18.helper/MainWindow.xaml.cs

```diff
@@ -0,0 +1,293 @@
+        public bool hasFile { get { return System.IO.File.Exists(jsonPath); } }
+        
+        public localizeReference(WeakReference<tsFileReference> parent, string country, string json, string code, string value)
+        {
+            jsonPath = json;
+            Parent = parent;
+            key = code;
+            cValue = value;
+            countryCode = country;
+        }
+
+        
+
+        public override string ToString()
+        {
+            return key;
+        }
+    }
+
+    public class tsFileReference
+    {
+        public string filePath { get; set; }
+        public string displayPath { get { return filePath.Replace(Shared.srcDir, ""); } }
+        public Dictionary<string, List<localizeReference>> References { get; set; } = new Dictionary<string, List<localizeReference>>();
+        public Dictionary<string, Dictionary<string, string>> jsonFiles = new Dictionary<string, Dictionary<string, string>>();
+        public tsFileReference(string tspath)
+        {
+            filePath = tspath;
+            List<string> Lines = System.IO.File.ReadAllLines(tspath).ToList();
+            foreach (var lin in Lines)
+            {
+                var parts = lin.Split(new string[] { "AutoLispExt.localize(" }, StringSplitOptions.None).ToList();
+                if (parts.Count > 1)
+                {
+                    bool stop = filePath.ToUpper().Contains("URI");
+                    int endindex = parts[1].IndexOf(',') == -1 ? parts[1].IndexOf(')') : parts[1].IndexOf(',');
+                    var localContent = parts[1].Substring(1, endindex - 1);
+                    //parts = localContent.Split(',').ToList();
+                    //if (parts.Count > 1)
+                    localContent = localContent.Trim().Trim('\'').Trim('\"').Trim('\'').Trim('\"');
+                    foreach (string langid in Shared.codes)
+                    {
+                        string expected = Shared.i18Dir + "\\" + langid + "\\out" + displayPath.Substring(0, displayPath.Length - 3) + ".i18n.json";
+                        if (jsonFiles.ContainsKey(expected) == false && System.IO.File.Exists(expected) == true)
+                            jsonFiles.Add(expected, expected.UnPack<Dictionary<string, string>>());
+                        else if (jsonFiles.ContainsKey(expected) == false)
+                            jsonFiles.Add(expected, new Dictionary<string, string>());
+
+                        string cv = jsonFiles[expected].ContainsKey(localContent) ? jsonFiles[expected][localContent] : "";
+
+                        if (References.ContainsKey(localContent) == false)
+                            References.Add(localContent, new List<localizeReference>());
+
+                        if (References[localContent].Count != Shared.codes.Count)
+                            References[localContent].Add(new localizeReference(new WeakReference<tsFileReference>(this), langid, expected, localContent, cv));
+                    }
```

Comment:
This is the section where the JSON files are either read using the UnPack extension method (if they exist and more specifically 265) into a simple `Dictionary<string, string>` or an empty `Dictionary<string, string>` is created (line 267) if they don't currently exist. IE, a new *.TS file was made with a brand new `AutoLispExt.localize()` reference. These values also have UI element representations (localizedReference objects) that will eventually overwrite the JSON Dictionary values using key value injection if the user clicks (line 106) save. I did it in this "cluster" of a way way so that only the found TypeScript references were ever injected and no orphaned or undetected localizations could ever be erased.